node/src/accountant/mod.rs

```diff
@@ -1742,129 +1688,105 @@ pub mod tests {
-            },
-            earning_wallet.clone(),
-        );
-
-        thread::spawn(move || {
-            let system =
-                System::new("accountant_logs_error_when_blockchain_bridge_responds_with_error");
-            let payable_dao = PayableDaoMock::new().non_pending_payables_result(vec![]);
-            let receivable_dao = ReceivableDaoMock::new()
-                .new_delinquencies_result(vec![])
-                .paid_delinquencies_result(vec![]);
-            let config_mock = PersistentConfigurationMock::new().start_block_result(Ok(0));
-            let subject = make_subject(
-                Some(config),
-                Some(payable_dao),
-                Some(receivable_dao),
-                None,
-                Some(config_mock),
-            );
-            let peer_actors = peer_actors_builder()
-                .blockchain_bridge(blockchain_bridge)
-                .build();
-            let subject_addr: Addr<Accountant> = subject.start();
-            let subject_subs = Accountant::make_subs_from(&subject_addr);
-
-            send_bind_message!(subject_subs, peer_actors);
-            send_start_message!(subject_subs);
-
-            system.run();
-        });
+        let system =
+            System::new("retrieve_transactions_sends_transactions_to_blockchain_interface");
+        let accountant = Recorder::new();
+        let earning_wallet = make_wallet("somewallet");
+        let expected_gas_price = 5u64;
+        let blockchain_interface_mock = BlockchainInterfaceMock::default()
+            .retrieve_transactions_result(Err(BlockchainError::QueryFailed(
+                "query failed, aborting message".to_string(),
+            )));
+        let persistent_configuration_mock =
+            PersistentConfigurationMock::default().gas_price_result(Ok(expected_gas_price));
+        let subject = BlockchainBridge::new(
+            Box::new(blockchain_interface_mock),
+            Box::new(persistent_configuration_mock),
+            false,
+            Some(earning_wallet.clone()),
+        );
+        let addr = subject.start();
+        let subject_subs = BlockchainBridge::make_subs_from(&addr);
+        let peer_actors = peer_actors_builder().accountant(accountant).build();
+        send_bind_message!(subject_subs, peer_actors);
+        System::current().stop();
+        system.run();
 
-        blockchain_bridge_awaiter.await_message_count(1);
-        let retrieve_transactions_recording = blockchain_bridge_recording.lock().unwrap();
-        let retrieve_transactions_message =
-            retrieve_transactions_recording.get_record::<RetrieveTransactions>(0);
-        assert_eq!(earning_wallet, retrieve_transactions_message.recipient);
+        addr.try_send(RetrieveTransactions {
+            start_block: 0u64,
+            recipient: earning_wallet.clone(),
+        })
+        .expect("query failed, aborting message");
```

Comment:
Sorry, this is really bad.

It's a misunderstanding of the concept of testing...
You cannot end the test prematurely and act as if it was no problem. 
See that the test aims at asserting around the logging and the assertion lies exactly at the very end of the test, but you panics earlier, before you get as far as where the assertion lies. 

I don't know what is worse:
a) you really don't assert at anything relevant in this test (because you don't start making the assertion)
b) you have a weird panic in the middle of the test and you don't care (and you instead rewrite the test so that it accepts the panic and looks green and fine)

You will have to try harder here. 

Hint: you cannot send messages after you shut down the system, which means after 

```
System::current().stop();
system.run();
```